src/core/public/http/http_service.mock.ts

```diff
@@ -37,7 +37,10 @@ const createServiceMock = ({ basePath = '' } = {}): ServiceSetupMockType => ({
   delete: jest.fn(),
   options: jest.fn(),
   basePath: new BasePath(basePath),
-  anonymousPaths: new AnonymousPaths(new BasePath(basePath)),
+  anonymousPaths: {
+    register: jest.fn(),
+    isAnonymous: jest.fn(),
```

Comment:
plugin authors doesn't have to read AnonymousPaths implementation to be able to use mocks. WDYT?